readline: Render completions using the entire line

daniel-shimon opened the pull request:
Since the completions we return are completed wrt. the completion prefix,
in order to handle spaced strings we need to use the entire line as the completion prefix.
For example:
    ls 'spaced
Can now correctly result in:
    ls 'spaced file' / ls 'spaced dir/'

Closes #2865

<!--- Thanks for opening a PR on xonsh! Please include a news entry with your PR
to help keep our changelog up to date! There are instructions available here:
https://xon.sh/devguide.html#changelog -->

<!--- If there is specific issue / feature request that this PR is addressing,
please link to the corresponding issue by using the `#issuenumber` syntax.
Thanks again! -->

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

jnoortheen commented on xonsh/readline_shell.py:
```diff
@@ -453,7 +453,7 @@ def completedefault(self, prefix, line, begidx, endidx):
             cursor_index=len(prev_text) + endidx,
         )
-        rtn_completions = _render_completions(completions, prefix, plen)
+        rtn_completions = _render_completions(completions, line, plen)
```
Is it possible to have tests for the rl-completions ? mocking could work or you can use `rl_shell` mock in `test_readline_shell.py` for testing.

daniel-shimon commented:
I'll look into it 👍

On Thu, Sep 9, 2021, 08:12 Noorhteen Raja NJ ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In xonsh/readline_shell.py
> <https://github.com/xonsh/xonsh/pull/4456#discussion_r704957567>:
>
> > @@ -453,7 +453,7 @@ def completedefault(self, prefix, line, begidx, endidx):
>              multiline_text=prev_text + line,
>              cursor_index=len(prev_text) + endidx,
>          )
> -        rtn_completions = _render_completions(completions, prefix, plen)
> +        rtn_completions = _render_completions(completions, line, plen)
>
> Is it possible to have tests for the rl-completions ? mocking could work
> or you can use rl_shell mock in test_readline_shell.py for testing.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xonsh/xonsh/pull/4456#pullrequestreview-749834924>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AELF3BJKEHIAAF425VTFLODUBA63DANCNFSM5DKBHSXA>
> .
>

gforsyth reviewed (approved):
The change looks good to me, I'll hold off on merging in case @daniel-shimon is planning to add some mocks for this

gforsyth commented:
Hey @daniel-shimon -- gentle ping here.  Is there anything else you wanted to add to this PR?

daniel-shimon commented:
> Hey @daniel-shimon -- gentle ping here.  Is there anything else you wanted to add to this PR?

Hi, sorry recently I don't have a lot of spare time to do xonsh stuff..

This PR has some problems which I addressed locally and added some half-baked tests.
I need to work out some details when I'll get to it.. For now this should be treated as a draft

gforsyth commented:
Closing out old PRs, thanks!
